src/main/java/net/dries007/tfc/compat/jade/common/BlockEntityTooltips.java

```diff
@@ -0,0 +1,332 @@
+        if (entity instanceof CropBlockEntity crop && state.getBlock() instanceof CropBlock block)
+        {
+            hoeOverlay(level, block, entity, tooltip);
+
+            tooltip.accept(Helpers.translatable("tfc.jade.yield").append(Helpers.literal(String.format("%.2f", crop.getYield()))));
+        }
+    };
+
+    public static final BlockEntityTooltip CRUCIBLE = (level, state, entity, tooltip) -> {
+        if (entity instanceof CrucibleBlockEntity crucible)
+        {
+            crucible.getCapability(HeatCapability.BLOCK_CAPABILITY).ifPresent(cap -> heat(tooltip, cap.getTemperature()));
+        }
+    };
+
+    public static final BlockEntityTooltip FIREPIT = (level, state, entity, tooltip) -> {
+        if (entity instanceof AbstractFirepitBlockEntity<?> firepit)
+        {
+            heat(tooltip, firepit.getTemperature());
+
+            if (firepit instanceof PotBlockEntity pot)
+            {
+                if (pot.shouldRenderAsBoiling())
+                {
+                    tooltip.accept(Helpers.translatable("tfc.tooltip.pot_boiling"));
+                }
+                else if (pot.getOutput() != null && !pot.getOutput().isEmpty())
+                {
+                    tooltip.accept(Helpers.translatable("tfc.tooltip.pot_finished"));
+
+                    if (pot.getOutput() instanceof SoupPotRecipe.SoupOutput soup)
+                    {
+                        final ItemStack stack = soup.stack();
+                        itemWithCount(tooltip, stack);
+
+                        final List<Component> text = new ArrayList<>();
+                        stack.getCapability(FoodCapability.CAPABILITY).ifPresent(cap -> cap.addTooltipInfo(stack, text));
+                        text.forEach(tooltip);
+                    }
+                }
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip FRUIT_TREE_SAPLING = (level, state, entity, tooltip) -> {
+        if (entity instanceof TickCounterBlockEntity counter && state.getBlock() instanceof FruitTreeSaplingBlock sapling)
+        {
+            final String growth = Helpers.formatPercentage(Math.min(0.99f, (float) counter.getTicksSinceUpdate() / (sapling.getTreeGrowthDays() * ICalendar.TICKS_IN_DAY)) * 100);
+            tooltip.accept(Helpers.translatable("tfc.jade.growth").append(growth));
+            hoeOverlay(level, sapling, entity, tooltip);
+        }
+    };
+
+    public static final BlockEntityTooltip HOE_OVERLAY = (level, state, entity, tooltip) -> {
+        if (state.getBlock() instanceof HoeOverlayBlock overlay)
+        {
+            hoeOverlay(level, overlay, entity, tooltip);
+        }
+    };
+
+    public static final BlockEntityTooltip LAMP = (level, state, entity, tooltip) -> {
+        if (entity instanceof LampBlockEntity lamp && state.getBlock() instanceof LampBlock)
+        {
+            // todo: this info is not available on the client (?)
```

Comment:
Please fix. Lamp fuels are currently not synced to client (can tell in `LampFuel#MANAGER` is missing arguments for encode / decode).